### PR TITLE
[GCI-258] Add relevant icons to OpenMRS AddOns modules

### PR DESCRIPTION
**Pull request by tomekdo3**
### General Description
I've added icons for modules with tags. To be sure that icons are relevant I've read module description and tags.
#### module.path, icon-name because reason
org.openmrs.module.appointmentscheduling, calendar because scheduling  related
**[removed]** ~~org.openmrs.module.appui, code because framework back-end related~~
org.openmrs.module.conceptsearch, search because search related
org.openmrs.module.drawing, picture-o because picture related
org.openmrs.module.filebrowser, search because search related
**[changed]** org.openmrs.module.formdataexport, share-square-o because export related
org.openmrs.module.formentry, pencil-square-o because form related
org.openmrs.module.formfilter, filter because filter related
org.openmrs.module.gmapsimageviewer, picture-o because picture related
org.openmrs.module.htmlformentryui, code because html related
org.openmrs.module.htmlformflowsheet, code because html related
org.openmrs.module.jasperreport, bar-chart because report related
org.openmrs.module.logic, superscript (x^2) math equation are associated with logic
org.openmrs.module.logicws, superscript (x^2) math equation are associated with logic
org.openmrs.module.logmanager, file-text-o because logging related
org.openmrs.module.messaging, comment-o because message related
**[changed]** org.openmrs.module.metadatadeploy, database because data related 
**[removed]** ~~org.openmrs.module.metadatamapping , database because data related~~
org.openmrs.module.metadatasharing, share-alt because sharing related
org.openmrs.module.mirthmessaging, comment-o because message related
org.openmrs.module.namephonetics, search because searching related
**[removed]** ~~org.openmrs.module.odamocklogicws, file-code-o because code and values related~~
org.openmrs.module.openhmis-backboneforms, code because backbone.js related
org.openmrs.module.patientimage, user-circle-o because patient picture related
**[changed]** org.openmrs.module.patientmatching, link because matching related
**[changed]** org.openmrs.module.registration, user-plus because registration related
org.openmrs.module.remoteformentry, pencil-square-o because form related
org.openmrs.module.reportingcompatibility, bar-chart because reporting related
org.openmrs.module.reportingrest, bar-chart because reporting related
org.openmrs.module.rolebasedhomepage, users because user role related
org.openmrs.module.schedulerquartz, calendar because scheduling related
**[changed]** org.openmrs.module.smsmodule, bell-o because message related
**[removed]** ~~org.openmrs.module.uicommons, code because back-end code related~~
org.openmrs.module.uiframework, code because framework code back-end related
org.openmrs.module.usagestatistics, bar-chart because statistics related
org.openmrs.module.visitscheduler, calendar because scheduling related
org.openmrs.module.chart-search-module,search because search related
org.openmrs.module.form-entry-app-module, pencil-square-o because form related
org.openmrs.module.media-viewer-module, picture-o because picture related
**[changed]** org.openmrs.module.appointment-scheduling-ui-module, desktop because UI related
**[changed]** org.openmrs.module.registration-core-module, user-plus because registration related
**[changed]** org.openmrs.module.registration-app-module, user-plus because registration related
**[changed]** org.openmrs.module.reporting-ui-module, desktop because UI related
**[changed]** org.openmrs.module.xreports, print because report generating related
org.openmrs.module.audit-log-module, file-text-o because logging related
org.openmrs.module.clinical-images, picture-o because picture related

### CODE REVIEW

**Comment by tomekdo3** [comment]
Hi @reubenvarghese1 I have a question. If there will be any conflicts with pull request by f4ww4z https://github.com/openmrs/openmrs-contrib-addonindex/pull/31 how it is gonna be solved? Would you mind looking at this?

**Comment by reubenvarghese1** [comment]
@tomekdo3  Sorry for delaying your reviews(was kinda caught up with something). As far as the merge conflicts are concerned, they are caused because both your PRs try to modify the same region of code. So, to fix it,I'll first review one of the PRs( In this case it will be @f4ww4z 's PR as he submitted it first). If I find any issue with it then I'll notify him; If no issue is found, I will merge his and then review yours following which , you will have to just pull the code and re-submit it based on the feedback received. Again, I regret the inconvenience caused to both of you

**Comment by reubenvarghese1** [comment]
Just reviewed @f4ww4z 's PR. Looks like quite a modules have been addressed by both of you; So, to fix the merge conflict. I will review each of your PRs individually as though the other doesn't exist yet. Once my review is complete. The one who submits a "merge-ready" PR first will get his PR merged while the other will just have to delete the ones already addressed by the merged commit and submit his existing unique changes. Hope this sounds fair.

File: src/main/resources/add-ons-to-index.json [hunk 1]
```diff
@@ -228,6 +229,7 @@
       "tags": [
         "back-end"
       ],
+      "icon": "code",
```
Comment (reubenvarghese1) [inline src/main/resources/add-ons-to-index.json]: Kindly delete/replace this as it looks irrelevant

File: src/main/resources/add-ons-to-index.json [hunk 2]
```diff
@@ -1058,6 +1063,7 @@
       "tags": [
         "form-entry"
       ],
+      "icon": "floppy-o",
```
Comment (reubenvarghese1) [inline src/main/resources/add-ons-to-index.json]: Kindly delete/replace this as it looks irrelevant

**Comment by tomekdo3** [comment]
@reubenvarghese1 I've removed one icon and changed floppy-o to share-square-o

**Comment by reubenvarghese1** [comment]
@tomekdo3  That was quick, I'm sorry but i had not completed the review.Will click on the more work needed icon in the dashboard once done reviewing

**Comment by tomekdo3** [comment]
Okey. Looking foward to it.

File: src/main/resources/add-ons-to-index.json [hunk 3]
```diff
@@ -1833,6 +1848,7 @@
       "tags": [
         "metadata"
       ],
+      "icon": "database",
```
Comment (reubenvarghese1) [inline src/main/resources/add-ons-to-index.json]: Kinda irrelevant. Should be deleted/replaced

Comment (reubenvarghese1) [reply src/main/resources/add-ons-to-index.json]: Could use "tag" or "tags"

File: src/main/resources/add-ons-to-index.json [hunk 4]
```diff
@@ -1858,6 +1874,7 @@
...
```
Comment (reubenvarghese1) [inline src/main/resources/add-ons-to-index.json]: Kinda irrelevant. Should be deleted/replaced

File: src/main/resources/add-ons-to-index.json [hunk 5]
```diff
@@ -2110,6 +2130,7 @@
       "tags": [
         "logic"
       ],
+      "icon": "file-code-o",
```
Comment (reubenvarghese1) [inline src/main/resources/add-ons-to-index.json]: Kinda irrelevant. Should be deleted/replaced

File: src/main/resources/add-ons-to-index.json [hunk 6]
```diff
@@ -2279,6 +2302,7 @@
       "tags": [
         "data-management"
       ],
+      "icon": "puzzle-piece",
```
Comment (reubenvarghese1) [inline src/main/resources/add-ons-to-index.json]: Good choice but I feel "link" is a better option as it represents the utility of the module better. Kindly check https://wiki.openmrs.org/display/docs/Patient+Matching+Module for more insight

File: src/main/resources/add-ons-to-index.json [hunk 7]
```diff
@@ -2519,6 +2543,7 @@
       "tags": [
         "registration"
       ],
+      "icon": "pencil-square-o",
```
Comment (reubenvarghese1) [inline src/main/resources/add-ons-to-index.json]: user-plus seems like a better option

File: src/main/resources/add-ons-to-index.json [hunk 8]
```diff
@@ -2987,6 +3017,7 @@
       "tags": [
         "messaging"
       ],
+      "icon": "comment-o",
```
Comment (reubenvarghese1) [inline src/main/resources/add-ons-to-index.json]: I would probably replace this with bell-o as it fits better(check out the wiki of the module for more insight)

File: src/main/resources/add-ons-to-index.json [hunk 9]
```diff
@@ -3833,6 +3870,7 @@
       "tags": [
         "scheduling"
       ],
+      "icon": "calendar",
```
Comment (reubenvarghese1) [inline src/main/resources/add-ons-to-index.json]: Adding desktop makes more sense as this module is primarily concerned with UI

File: src/main/resources/add-ons-to-index.json [hunk 10]
```diff
@@ -3855,6 +3893,7 @@
       "tags": [
         "registration"
       ],
+      "icon": "pencil-square-o",
```
Comment (reubenvarghese1) [inline src/main/resources/add-ons-to-index.json]: user-plus

File: src/main/resources/add-ons-to-index.json [hunk 11]
```diff
@@ -3877,6 +3916,7 @@
...
```
Comment (reubenvarghese1) [inline src/main/resources/add-ons-to-index.json]: user-plus

File: src/main/resources/add-ons-to-index.json [hunk 12]
```diff
@@ -3966,6 +4006,7 @@
       "tags": [
         "reporting"
       ],
+      "icon": "bar-chart",
```
Comment (reubenvarghese1) [inline src/main/resources/add-ons-to-index.json]: desktop

File: src/main/resources/add-ons-to-index.json [hunk 13]
```diff
@@ -4220,6 +4261,7 @@
...
```
Comment (reubenvarghese1) [inline src/main/resources/add-ons-to-index.json]: Irrelevant as this module not exactly modifying reports. Kindly check its usage and delete/modify this accordingly

**Comment by tomekdo3** [comment]
@reubenvarghese1 I think that's it. I've removed two icons and changed the rest according to your suggestions + changed xreports to print icon. What do you think about that?

**Comment by reubenvarghese1** [comment]
@tomekdo3 Looks good to me now! Good effort, I will approve this task. I will merge this once we get a go ahead from @djazayeri as well

**Comment by tomekdo3** [comment]
Thank you!

**Comment by djazayeri** [comment]
@reubenvarghese1 sorry for missing this while I was out on Christmas break. I trust your judgment on this, so if you think it's good to merge, go ahead!

(And also you should promote the new version from staging to production. Let me know if you don't have privs to do this from ci.openmrs.org.)

**Comment by reubenvarghese1** [comment]
Awesome, I'm merging this now. Thanks for working on this @tomekdo3 !
